Tests/ApolloTests/Network/URLSessionClientTests.swift

```diff
@@ -1,14 +1,18 @@
 import XCTest
 @testable import Apollo
+import ApolloInternalTestHelpers
 
-class URLSessionClientLiveTests: XCTestCase {
+class URLSessionClientTests: XCTestCase {
   
   var client: URLSessionClient!
+  var sessionConfiguration: URLSessionConfiguration!
 
   override func setUp() {
     super.setUp()
-
-    client = URLSessionClient()
+    Self.testObserver.start()
```

Comment:
Running the `testObserver.start()` in the setup for every test rather that on value change for `requestHandlers` like what is done in `MockShemaData` due to some tests applying more than 1 request handler, in testing this on top of the `MockURLProtocol` clearing out request handlers as it uses them ensured that there were no left over request handlers between tests.

Reply:
Yupp that's perfect!